lumicks/pylake/kymotracker/tests/test_io.py

```diff
@@ -0,0 +1,78 @@
+from lumicks.pylake.kymotracker.kymoline import KymoLine, KymoLines, import_kymolines_from_csv
+import numpy as np
+import pytest
+
+
+@pytest.fixture(scope="session")
+def kymolines_io_data():
+    test_img = np.zeros((8, 8))
+
+    k1 = KymoLine([1, 2, 3], [2, 3, 4], test_img)
+    k2 = KymoLine([2, 3, 4], [3, 4, 5], test_img)
+    k3 = KymoLine([3, 4, 5], [4, 5, 6], test_img)
+    k4 = KymoLine([4, 5, 6], [5, 6, 7], test_img)
+    lines = KymoLines([k1, k2, k3, k4])
+
+    for k in lines:
+        test_img[k.coordinate_idx, k.time_idx] = 2
+        test_img[np.array(k.coordinate_idx) - 1, k.time_idx] = 1
+
+    return test_img, lines
+
+
+def read_txt(testfile, delimiter):
+    raw_data = np.loadtxt(testfile, delimiter=delimiter, unpack=True)
+    with open(testfile, "r") as f:
+        data = {}
+        header = f.readline().rstrip().split(delimiter)
+        line_idx = raw_data[0, :]
+        for key, col in zip(header, raw_data):
+            data[key] = [col[np.argwhere(line_idx == idx).flatten()] for idx in np.unique(line_idx)]
+
+        return data
+
+
+@pytest.mark.parametrize("dt, dx, delimiter, sampling_width, sampling_outcome",
+                         [[1.0, 1.0, ';', 0, 2],
+                          [2.0, 1.0, ';', 0, 2],
+                          [1.0, 2.0, ';', 0, 2],
+                          [1.0, 1.0, ',', 0, 2],
+                          [1.0, 1.0, ';', 1, 3],
+                          [None, None, ';', 1, 3],
+                          [None, None, ';', None, 3],
+                          [1.0, 2.0, ';', None, None],
+                          [None, 1.0, ';', None, None],
+                          [1.0, None, ';', None, None]])
+def test_kymolines_io(tmpdir_factory, kymolines_io_data, dt, dx, delimiter, sampling_width, sampling_outcome):
+    test_img, lines = kymolines_io_data
+
+    # Test round trip through the API
+    testfile = f"{tmpdir_factory.mktemp('pylake')}/test.csv"
+    lines.save(testfile, dt, dx, delimiter, sampling_width)
+    read_file = import_kymolines_from_csv(testfile, test_img, delimiter=delimiter)
+
+    # Test raw fields
+    data = read_txt(testfile, delimiter)
+
+    for l1, l2 in zip(lines, read_file):
```

Comment:
nitpick: I'm all for using short variable names, but I really try to avoid using a single `l`. Personally I would prefer `line1` and `line2` instead?

Reply:
I only do it when the scope is one or two lines, but I'm alright with avoiding it. I'll change it to `line1` and `line2` :)